drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -14,41 +14,34 @@
 LOG_MODULE_REGISTER(pwm_nrf5_sw);
 
 #define TIMER_INSTANCE  _CONCAT(TIMER, DT_NORDIC_NRF_SW_PWM_0_TIMER_INSTANCE)
+#define TIMER_REGS      _CONCAT(NRF_, TIMER_INSTANCE)
 #define GET_TIMER_CAPABILITY(capability) \
 	(_CONCAT(_CONCAT(TIMER_INSTANCE, _), capability))
 
 /* One compare channel is needed to set the PWM period, hence +1. */
 #if ((DT_NORDIC_NRF_SW_PWM_0_CHANNEL_COUNT + 1) > GET_TIMER_CAPABILITY(CC_NUM))
 #error "Invalid number of PWM channels configured."
 #endif
-#define PWM_0_MAP_SIZE DT_NORDIC_NRF_SW_PWM_0_CHANNEL_COUNT
+#define PWM_MAP_SIZE  DT_NORDIC_NRF_SW_PWM_0_CHANNEL_COUNT
 
 /* Nordic TIMER peripherals allow prescalers 0-9. */
 #define MAX_TIMER_PRESCALER  9
 /* Nordic TIMERs can be 16- or 32-bit wide. */
 #define MAX_TIMER_VALUE  (GET_TIMER_CAPABILITY(MAX_SIZE) == 32 ? UINT32_MAX \
 							       : UINT16_MAX)
 
-struct pwm_config {
```

Comment:
Can the hardware support multiple instances? If so, this is a step in the wrong direction. Why not fix the driver to use the device tree data consistently?

Reply:
The driver actually combines several peripherals to achieve the PWM functionality. But yes, it can be made multi-instance (though, due to the checks of capabilities of the selected TIMER, this will probably require more macro magic). I was not sure if it would be desirable and didn't want to do this in this PR. The removal of this structure was to clean up the current state (the driver was single-instance from its origin).